Scripts/Menu System/MenuSystem.gd

```diff
@@ -1,79 +1,83 @@
 extends Node
 class_name MenuSystem
 
+## To disable the menu system when working on other systems.
 @export var disable : bool = false
 
-# Add values to this enum for add extra menus, also InGame menu is supposed to be a blank scene
-# Health bar, xp level labels should be its own canvas layer and not attached to the menu manager
-enum Keys {Play,MainMenu,Paused,Settings}
-# Storing the menus in a dict using the menu keys as a way to index it.
+## Add to the keys enum when adding a new menu.
+enum Keys {InGame,MainMenu,Paused,Settings,Credits}
+## Stores all the menus in memory and stored in a dict using the keys enum as input.
 var menus : Dictionary = {}
-# Current menu is the actual scene
+## Current menu that is being shown.
 var current_menu : Control
-var current_menu_key : Keys
-# Previous menus is how we load previous menus so the current menu doesnt need
-# to have information about previous menus
+## Storing a single array of menus so we can go back a menu without the new menu knowing it.
 var previous_menus : Array = []
-# Where we instance the current menu
+## Where menus are instanced into the scene.
 var menu_container : CanvasLayer
 
 func _ready() -> void:
 	
 	if disable: return
 	
+	## Need to create the container first before we event start loading menus.
 	set_up_menu_container()
 	
-	add_menu(Keys.Play,"res://Menus/Play/PlayMenu.tscn")
-	add_menu(Keys.Paused,"res://Menus/PauseMenu/pause_menu.tscn")
-	add_menu(Keys.MainMenu,"res://Menus/MainMenu/main_menu.tscn")
-	add_menu(Keys.Settings,"res://Menus/Settings/Settings.tscn")
-	
+	## Adding all the menus we want the game to have in the ready func.
+	add_menu(Keys.InGame,"res://Scenes/Menus/in_game_menu.tscn")
+	add_menu(Keys.Paused,"res://Scenes/Menus/pause_menu.tscn")
+	add_menu(Keys.MainMenu,"res://Scenes/Menus/main_menu.tscn")
+	add_menu(Keys.Settings,"res://Scenes/Menus/settings_menu.tscn")
+	add_menu(Keys.Credits,"res://Scenes/Menus/credits_menu.tscn")
+	## When all the menus are setup we now need to load the first menu. 
 	load_menu(Keys.MainMenu)
 
+## Adding the new menu into the dict and is loaded into memory.
 func add_menu(key : Keys , menu_path : String) -> void:
 	menus[key] = load(menu_path)
```

Comment:
This is... odd. What I'm seeing here is that you're preloading all the scenes in advance. While this does have it's merits, it will cause issues once we start adding more to our scenes. If you could explain what your approach is here that would be amazing!

Reply:
Im not sure what issues could arise from this but its simply like this, load all the menus into a dictionary and then I can instance each menu into a canvas layer node, and its simply queue freeing previous menu and instancing the new menu, its simple as that, also each menu tells the menu system which menu to load next while also keeping track of previous menus so we can go back without the menu needing to know other menus.

Reply:
It's certainly an option, and will be useful when we have a lot more menus, but for now dynamic loading/unloading of menus could cause issues. It may be a better idea to, while we don't have a large amount of menus to keep track of, have them all exist at once in a TabContainer or otherwise. I leave it up to you but it feels unnecessary.